src/main/java/uk/gov/digital/ho/proving/income/validator/CatBSalariedIncomeValidator.java

```diff
@@ -87,15 +56,41 @@ private IncomeValidationResult validateForIndividual(IncomeValidationRequest inc
         return validationResult(incomeValidationRequest, IncomeValidationStatus.CATB_SALARIED_PASSED);
     }
 
+    private List<List<Income>> sortAndGroupIncomesByMonth(List<Income> incomes) {
+        List<List<Income>> monthlyIncomes = new ArrayList<>();
+        incomes.stream()
+            .collect(Collectors.groupingBy(Income::yearAndMonth))
+            .forEach((yearAndMonth, income) -> monthlyIncomes.add(income));
+
+        monthlyIncomes.sort(Comparator.comparingInt(monthlyIncome -> monthlyIncome.get(0).yearAndMonth()));
+        return monthlyIncomes;
+    }
+
+    private List<Income> getAllPayeInDateRange(IncomeValidationRequest incomeValidationRequest) {
+        List<Income> paye = getAllPayeIncomes(incomeValidationRequest);
+        LocalDate applicationStartDate = getApplicationStartDate(incomeValidationRequest);
+        LocalDate applicationRaisedDate = incomeValidationRequest.applicationRaisedDate();
+        return filterIncomesByDates(paye, applicationStartDate, applicationRaisedDate)
+            .collect(Collectors.toList());
+    }
+
+    private List<Income> getAllPayeIncomes(IncomeValidationRequest incomeValidationRequest) {
```

Comment:
This code is duplicated in CatBNonSalariedIncomeValidator.  Can we also move this to either IncomeValidationRequest or IncomeValidationHelper?

Reply:
Agreed